istioctl/cmd/istioctl/kubeinject.go

```diff
@@ -261,46 +259,7 @@ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConf
 			}
 
 			var sidecarTemplate string
-
-			// hub and tag params only work with ISTIOCTL_USE_BUILTIN_DEFAULTS
-			// so must be specified together. hub and tag no longer have defaults.
-			if hub != "" || tag != "" {
-				// ISTIOCTL_USE_BUILTIN_DEFAULTS is used to have legacy behavior.
-				if !getBoolEnv("ISTIOCTL_USE_BUILTIN_DEFAULTS", false) {
-					return errors.New("one of injectConfigFile or injectConfigMapName is required\n" +
-						"use the following command to get the current injector file\n" +
-						"kubectl -n istio-system get configmap istio-sidecar-injector " +
-						"-o=jsonpath='{.data.config}' > /tmp/injectConfigFile.yaml")
-				}
-
-				if hub == "" || tag == "" {
-					return fmt.Errorf("hub and tag are both required. got hub: '%v', tag: '%v'", hub, tag)
-				}
-
-				if sidecarTemplate, err = inject.GenerateTemplateFromParams(&inject.Params{
-					InitImage:                    inject.InitImageName(hub, tag, debugMode),
-					ProxyImage:                   inject.ProxyImageName(hub, tag, debugMode),
-					RewriteAppHTTPProbe:          rewriteAppHTTPProbe,
-					Verbosity:                    verbosity,
-					SidecarProxyUID:              sidecarProxyUID,
-					Version:                      versionStr,
-					EnableCoreDump:               enableCoreDump,
-					Mesh:                         meshConfig,
-					ImagePullPolicy:              imagePullPolicy,
-					StatusPort:                   statusPort,
-					ReadinessInitialDelaySeconds: readinessInitialDelaySeconds,
-					ReadinessPeriodSeconds:       readinessPeriodSeconds,
-					ReadinessFailureThreshold:    readinessFailureThreshold,
-					IncludeIPRanges:              includeIPRanges,
-					ExcludeIPRanges:              excludeIPRanges,
-					IncludeInboundPorts:          includeInboundPorts,
-					ExcludeInboundPorts:          excludeInboundPorts,
-					DebugMode:                    debugMode,
-				}); err != nil {
-					return err
-				}
-
-			} else if injectConfigFile != "" {
+			if injectConfigFile != "" {
```

Comment:
You should remove or throw an error with this change. Otherwise, someone could specify a flag and it would be ignored.

Reply:
At present, these flags are already be ignored, unless set `ISTIOCTL_USE_BUILTIN_DEFAULTS=1`.

Reply:
Looking at the right code, there exists many inject related overrides that aren't hooked up to anything. Some of this will print a deprecation warning which is no longer accurate. If support for the build in template is removed, we should also remove all of those flags. This will produce an error (e.g. unknown flag) if users were previously depending on them which is exactly the behavior we want.